### DISCUSSION TITLE
Dewheatertest.py error

### DISCUSSION THREAD

**Discussion by SkyDragon2023** [Curtis2012/dewheater#14] [Q&A]
I don't now what I've missed but I managed to download the files, I've checked that my BME280 sensor is detected, but when I run the dewheatertest.py I get the following:
pi@allsky:~/dewheater $ sudo python dewheatertest.py
Traceback (most recent call last):
  File "/home/pi/dewheater/dewheatertest.py", line 13, in <module>
    from meteocalc import dew_point
ModuleNotFoundError: No module named 'meteocalc'
How do I fix this?

**Comment by Curtis2012** [comment]
No need to run it under sudo. 

Looks like you need to install the "meteocalc" package. 

https://pypi.org/project/meteocalc/

**Reply by SkyDragon2023** [reply]
Ok thank you, I didn't know I need that installed. Now I get the following error:
pi@allsky:~/dewheater $ python dewheatertest.py
Traceback (most recent call last):
  File "/home/pi/dewheater/dewheatertest.py", line 16, in <module>
    import bme280
  File "/home/pi/dewheater/bme280.py", line 162
    print "Chip ID     :", chip_id
          ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Chip ID     :", chip_id)?

**Reply by SkyDragon2023** [reply]
Well fixed one error, missing ( ) around print commands at the end of the bme280.py lines.
Now I get the following:
pi@allsky:~/dewheater $ python dewheatertest.py
Traceback (most recent call last):
  File "/home/pi/dewheater/dewheatertest.py", line 25, in <module>
    calibration_params = bme280.load_calibration_params(bus, address)
AttributeError: module 'bme280' has no attribute 'load_calibration_params'

If this has been working for other people why is this not working correctly on my Raspberry Pi? it's a stock install of the latest of the Pi imager, running TJ's Allsky software.

**Reply by Curtis2012** [reply]
Thats odd that you would get a syntax error. Somehow you have an incorrect/inwork version of the code. 

Which branch did you install from?

**Reply by SkyDragon2023** [reply]
I think I ran git clone --recursive https://github.com/Curtis2012/dewheater.git
I'm fairly new to raspberry pi software loading.

**Reply by Curtis2012** [reply]
Clone from which branch though? If you are getting syntax errors then it is likely from a branch that is in work, not the master branch. The master branch does not even contain the code referenced by your error message. 

I suggest you just use the github website interface and clone the master branch.

**Reply by Curtis2012** [reply]
> Ok thank you, I didn't know I need that installed. Now I get the following error: pi@allsky:~/dewheater $ python dewheatertest.py Traceback (most recent call last): File "/home/pi/dewheater/dewheatertest.py", line 16, in import bme280 File "/home/pi/dewheater/bme280.py", line 162 print "Chip ID :", chip_id ^ SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Chip ID :", chip_id)?

Note that bme280.py is not part of the dewheater code. Looks like you may have an issue with the BME module. 

Look at the installnotes.txt in the dewheater master repository. It lists the modules that need to be installed.

**Reply by Curtis2012** [reply]
> Well fixed one error, missing ( ) around print commands at the end of the bme280.py lines. Now I get the following: pi@allsky:~/dewheater $ python dewheatertest.py Traceback (most recent call last): File "/home/pi/dewheater/dewheatertest.py", line 25, in calibration_params = bme280.load_calibration_params(bus, address) AttributeError: module 'bme280' has no attribute 'load_calibration_params'
> 
> If this has been working for other people why is this not working correctly on my Raspberry Pi? it's a stock install of the latest of the Pi imager, running TJ's Allsky software.

Looks like you may be picking up the wrong BME280 module. There are several out there. Again, see installnotes.txt.

**Reply by SkyDragon2023** [reply]
I only installed the files listed in the install notes using the commands given, so I can't have installed the wrong ones.  I have tested the sensor with code from there page (https://pypi.org/project/RPi.bme280/) and it gives readings, so as far as I know the sensor is working.
7e142c48-594d-46cd-b7c9-f54caa6aa9e5
2023-04-10 09:17:30.023306+00:00
27.95819312578533
996.0798343154867
33.062146225041836
compensated_reading(id=7e142c48-594d-46cd-b7c9-f54caa6aa9e5, timestamp=2023-04-10 09:17:30.023306UTC, temp=27.958 °C, pressure=996.08 hPa, humidity=33.06 % rH)

**Comment by SkyDragon2023** [comment]
I deleted all the files in the dewheater folder, download the zip file at the top of the page, unpacked them to my pi. I run the dewheatertest.py file and get the following:
pi@allsky:~/dewheater $ python dewheatertest.py
{
    "bmeAddress": "0x76",
    "bmePort": 1,
    "debug": true,
    "dewHeaterCutinOffset": 1.0,
    "dewHeaterCutoutOffset": 1.0,
    "dewHeaterMaxTemp": 30.0,
    "dewHeaterMaxTimeOn": 6.5,
    "dewHeaterMinTemp": 1.0,
    "dewHeaterOnOffDelay": 5,
    "dewHeaterPin": 23,
    "dewPtCheckDelay": 5,
    "dhtPin": 4,
    "fakeDewPoint": false,
    "fakeDewPointSamples": 20,
    "invertOnOff": false,
    "sensorType": "BME280"
}
/home/pi/dewheater/dewheatertest.py:45: RuntimeWarning: This channel is already in use, continuing anyway.  Use GPIO.setwarnings(False) to disable warnings.
  GPIO.setup(config.dewHeaterPin, GPIO.OUT)
dew-heater-test started.
Traceback (most recent call last):
  File "/home/pi/dewheater/dewheatertest.py", line 98, in <module>
    main()
  File "/home/pi/dewheater/dewheatertest.py", line 89, in main
    dewHeater.checkTemp()   # read and display temps for reference
  File "/home/pi/dewheater/dewheatertest.py", line 80, in checkTemp
    print("Temp = %3.1fC Humidity %3.1f%%" % (temperature, humidity))
NameError: name 'temperature' is not defined
pi@allsky:~/dewheater $

I made sure the following were installed:
install RPi.bme280
install bme208:
install smbus2

**Reply by Curtis2012** [reply]
Please post the zip file you downloaded.

**Reply by Curtis2012** [reply]
Replace  your dewheatertest.py file with the file at the link below and see if that works for you. No DropBox account is needed. 

https://www.dropbox.com/s/z26uv0l4vhvhg1w/dewheatertest.py?dl=0

**Reply by SkyDragon2023** [reply]
Download and ran the file, same result: 
`pi@allsky:~/dewheater $ python dewheatertest.py
{
    "bmeAddress": "0x76",
    "bmePort": 1,
    "debug": true,
    "dewHeaterCutinOffset": 1.0,
    "dewHeaterCutoutOffset": 1.0,
    "dewHeaterMaxTemp": 30.0,
    "dewHeaterMaxTimeOn": 6.5,
    "dewHeaterMinTemp": 1.0,
    "dewHeaterOnOffDelay": 5,
    "dewHeaterPin": 23,
    "dewPtCheckDelay": 5,
    "dhtPin": 4,
    "fakeDewPoint": false,
    "fakeDewPointSamples": 20,
    "invertOnOff": false,
    "sensorType": "BME280"
}
/home/pi/dewheater/dewheatertest.py:45: RuntimeWarning: This channel is already                                        in use, continuing anyway.  Use GPIO.setwarnings(False) to disable warnings.
  GPIO.setup(config.dewHeaterPin, GPIO.OUT)
dew-heater-test started.
Traceback (most recent call last):
  File "/home/pi/dewheater/dewheatertest.py", line 98, in <module>
    main()
  File "/home/pi/dewheater/dewheatertest.py", line 89, in main
    dewHeater.checkTemp()   # read and display temps for reference
  File "/home/pi/dewheater/dewheatertest.py", line 79, in checkTemp
    self.temperature = data.temerature
AttributeError: 'compensated_readings' object has no attribute 'temerature'
pi@allsky:~/dewheater $
`

**Reply by SkyDragon2023** [reply]
I think you have a spelling error, your missing a "p" in temperature

**Reply by SkyDragon2023** [reply]
Ok it's working once I corrected the spelling error with the missing "p"!

**Reply by Curtis2012** [reply]
Arrgh. Looks like an inwork version of dewheatertest.py must have go promoted to main. 

Sorry for the inconvenience.

**Comment by SkyDragon2023** [comment]
[dewheater-master.zip](https://github.com/Curtis2012/dewheater/files/11189341/dewheater-master.zip)
here is the file, downloaded from your githib page at the top of the homepage.

**Comment by Curtis2012** [comment]
@SkyDragon2023 

I promoted the correct dewheatertest.py to main, but I am on the road and cannot test to confirm. Please download and test one last time to confirm.
